server/src/models/roles/role.dto.ts

```diff
@@ -0,0 +1,70 @@
+import { Prisma, Role } from '@prisma/client';
+import {
+    IsBoolean,
+    IsNotEmpty,
+    IsString,
+    IsDefined,
+    IsNotEmptyObject,
+    IsObject,
+    ValidateNested,
+    
+} from 'class-validator';
+import { Type } from 'class-transformer';
+
+export class PermissionDto {
```

Comment:
Not an issue for you but just as a discussion starter, these validators are the number one thing we should be testing in unit tests. Just to keep it in mind, once we get the basic features shipped, we should start building tests for them ASAP.